pickhardtpayments/ChannelGraph.py

```diff
@@ -46,3 +53,80 @@ def get_channel(self, src: str, dest: str, short_channel_id: str):
         if self.network.has_edge(src, dest):
             if short_channel_id in self.network[src][dest]:
                 return self.network[src][dest][short_channel_id]["channel"]
+
+
+def lnd2cln_json(channel_graph_json):
+    """
+    Converts the channel graph json from the LND format to the c-lightning format
+    """
+
+    # Maps LND keys to CLN keys
+    LND_CLN_POLICY_MAP = {
+        "time_lock_delta": ChannelFields.CLTV,
+        "min_htlc": ChannelFields.HTLC_MINIMUM_MSAT,
+        "fee_base_msat": ChannelFields.BASE_FEE_MSAT,
+        "fee_rate_milli_msat": ChannelFields.FEE_RATE,
+        "disabled": ChannelFields.ACTIVE,
+        "max_htlc_msat": ChannelFields.HTLC_MAXIMUM_MSAT,
+        "last_update": ChannelFields.LAST_UPDATE,
+    }
+
+    def _add_direction(src, dest, lnd_channel, cln_channel):
+        cln_channel[ChannelFields.SRC] = lnd_channel[src + "_pub"]
+        cln_channel[ChannelFields.DEST] = lnd_channel[dest + "_pub"]
+
+        src_policy = lnd_channel[src + "_policy"]
+        for key in src_policy:
+            val = int(src_policy[key]) if key != "disabled" else src_policy[key]
+            cln_channel[LND_CLN_POLICY_MAP[key]] = val
+
+    def _find_node(pubkey, nodes_list):
+        for node in nodes_list:
```

Comment:
can't you initially just load all nodes into a `dict` with `pubkey` as index and then just fetch them instead of going linearly through the list everytime you need it?

Reply:
Done in 9c733e1